app/controllers/concerns/curation_concerns/curation_concern_controller.rb

```diff
@@ -194,7 +194,17 @@ def curation_concern_from_search_results
       def search_result_document(search_params)
         _, document_list = search_results(search_params)
         raise CanCan::AccessDenied.new(nil, :show) if document_list.empty?
+        if document_suppressed?(document_list)
+          flash[:notice] = I18n.t("curation_concerns.workflow.default.unauthorized")
+          raise CanCan::AccessDenied.new(nil, :show)
+        end
         document_list.first
       end
+
+      def document_suppressed?(document_list)
+        unless current_user
```

Comment:
This is not an anonymous user. If you are looking for anonymous users, you probably want to check for `guest?` Keep in mind that guest users exist and have `guest?` only if the `devise-guest` gem is being used, otherwise, you'd check for anonymous by checking `current_user.new_record?`